src/frontend/src/pages/myfeeds.js

```diff
@@ -15,94 +34,312 @@ const useStyles = makeStyles((theme) => ({
+            onClick={() => {
+              handleClose();
+              removeFeed();
+            }}
+            color="primary"
+            variant="contained"
+          >
+            Confirm
+          </Button>
+        </DialogActions>
+      </Dialog>
+    </div>
+  );
+}
+
+DeleteDialogButton.propTypes = {
+  feed: PropTypes.object,
+};
+
 export default function MyFeeds() {
   const classes = useStyles();
-  const [feedAuthor, setFeedAuthor] = useState('');
-  const [feedUrl, setFeedUrl] = useState('');
+  const [newFeedAuthor, setNewFeedAuthor] = useState('');
+  const [newFeedUrl, setNewFeedUrl] = useState('');
+  const [submitStatus, setSubmitStatus] = useState({ message: '', isError: false });
+  const [userInfo, setUserInfo] = useState({});
+  const [feedHash, updateFeedHash] = useState({}); // { id1: {author: '...', url: '...'} }, ... ]
+
   const { telescopeUrl } = useSiteMetadata();
 
-  function handleAuthorChange(author) {
-    setFeedAuthor(author);
-  }
+  const newFeedAuthorRef = React.createRef();
+  const newFeedUrlRef = React.createRef();
+
+  async function hashUserFeeds() {
+    const [user, feedItems] = await Promise.all([
+      fetch(`${telescopeUrl}/user/info`).then((res) => res.json()),
+      fetch(`${telescopeUrl}/feeds`).then((res) => res.json()),
+    ]);
+    const allFeeds = await Promise.all(
+      feedItems.map((item) => fetch(`${telescopeUrl}${item.url}`))
+    ).then((responses) => Promise.all(responses.map((res) => res.json())));
+
+    if (allFeeds.length && user) {
+      const userFeeds = allFeeds.filter((feed) => {
+        return feed.user === user.id;
+      });
 
-  function handleUrlChange(url) {
-    setFeedUrl(url);
+      const userFeedHash = userFeeds.reduce((hash, feed) => {
+        hash[feed.id] = { author: feed.author, url: feed.url };
+        return hash;
+      }, {});
+
+      return Promise.all([updateFeedHash(userFeedHash), setUserInfo(user)]);
+    }
+    return Promise.reject();
   }
 
+  useEffect(() => {
+    hashUserFeeds();
+    ValidatorForm.addValidationRule('isUrl', (value) => !!isWebUri(value));
+    return ValidatorForm.removeValidationRule.bind('isUrl');
+  }, []);
+
   async function addFeed() {
+    setSubmitStatus({ message: 'Adding new feed, please wait...' });
     try {
       const response = await fetch(`${telescopeUrl}/feeds`, {
         method: 'post',
         headers: {
           'Content-Type': 'application/json',
         },
         body: JSON.stringify({
-          author: feedAuthor,
-          url: feedUrl,
+          user: userInfo.id,
+          author: newFeedAuthor,
+          url: newFeedUrl,
         }),
       });
-      console.log(response);
-      return response;
+      await setSubmitStatus(
+        response.ok
+          ? { message: 'Feed added successfully' }
+          : { message: `Error: ${response.status} ${response.statusText}`, isError: true }
+      );
+      if (response.ok) {
+        window.location.reload(false);
```

Comment:
Why?

Reply:
Programmatically reloading the page was a quick and dirty way to reset form fields (`react-material-ui-form-validator` doesn't appear to offer an easy way to do so) while also forcing a rerender. I've now removed these refresh() calls in https://github.com/Seneca-CDOT/telescope/pull/933/commits/2ae7efe11944d53d23e43c897c561cf9948cfedb, having invested some time tonight becoming more comfortable with the `useEffect` hook.